gensim/models/word2vec.py

```diff
@@ -509,9 +510,16 @@ def build_vocab(self, sentences, keep_raw_vocab=False, trim_rule=None, progress_
         Each sentence must be a list of unicode strings.
 
         """
+        for sentence in sentences:
```

Comment:
Oh alright I didn't know about the all-or-nothing scenario. I've added checks in the `scan_vocab` methods of `Word2Vec` and `Doc2Vec` wherein I'm only checking if the `sentence` or `document.words` is an instance of `string_types` for sentence (or document) 0. So I don't we need the check for all single characters in `vocab` since, as you said, this check already covers it. Is this approach fine?